proxy/http/HttpTransact.cc

```diff
@@ -762,8 +762,8 @@ HttpTransact::StartRemapRequest(State *s)
 
   const char syntxt[] = "synthetic.txt";
 
-  s->cop_test_page = (ptr_len_cmp(host, host_len, local_host_ip_str, sizeof(local_host_ip_str) - 1) == 0) &&
-                     (ptr_len_cmp(path, path_len, syntxt, sizeof(syntxt) - 1) == 0) &&
+  s->cop_test_page = (host_len == sizeof(local_host_ip_str) - 1) && (path_len == sizeof(syntxt) - 1) &&
+                     (memcmp(host, local_host_ip_str, host_len) == 0) && (memcmp(path, syntxt, path_len) == 0) &&
```

Comment:
That feels that it's orthogonal from this Jira / PR, and could be moved to a different Jira. But up to Tyler to decide IMO.